learn-by-rewriting/es1/array-api/indexOf.js

```diff
@@ -26,11 +26,23 @@
   Link to docs https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexof
 */
 
-function indexOf(/*function parameters*/) {
-  
-  /*
-    Your code goes here
-  */
+function indexOf(array, searchElement, fromIndex=0) {
+/*
+  guideline recommended to _avoid using other array methods_
+  if( !Array.isArray(array) ) {
+    throw new TypeError('first argument is not an array');
+  }
+*/
+  if(typeof array[Symbol.iterator] !== 'function') {
+    throw new TypeError('first argument is not an iterable');
+  }
+  if( !isFinite(fromIndex) ) {
+    fromIndex = 0
+  }
+  for ( let index = fromIndex; index <= array.length; index++  ) {
```

Comment:
`index` would be `fromIndex` if positive
`index` would be `array.length + fromIndex` if negative. You can make index 0 if this offset is below zero. Not sure if that is necessery.

Reply:
I'm already handling the negative value in the `if` condition above just like you're mentioning here - i.e., set `fromIndex = 0`. Is that what you mean?

There were tests with negative `fromIndex` which forced me to write that check before I got to the `for` loop.

Reply:
I've added some checks for the negative `fromIndex` as per the [`indexOf` specification](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) per your suggestion